src/main/java/fr/moribus/imageonmap/gui/MapDetailGui.java

```diff
@@ -208,32 +209,35 @@ public void rename() {
             return;
         }
 
-        PromptGui.prompt(getPlayer(), newName -> {
-            if (!Permissions.RENAME.grantedTo(getPlayer())) {
-                I.sendT(getPlayer(), "{ce}You are no longer allowed to do that.");
-                return;
-            }
-
-            if (newName == null || newName.isEmpty()) {
-                I.sendT(getPlayer(), "{ce}Map names can't be empty.");
-                return;
-            }
-            if (newName.equals(map.getName())) {
-                return;
-            }
-
-            map.rename(newName);
-            I.sendT(getPlayer(), "{cs}Map successfully renamed.");
-
-            if (getParent() != null) {
-                RunTask.later(() -> Gui.open(getPlayer(), this), 1L);
-
-            } else {
-                close();
-            }
-        }, map.getName(), this);
-
-
+        try {
+            PromptGui.prompt(getPlayer(), newName -> {
+                if (!Permissions.RENAME.grantedTo(getPlayer())) {
+                    I.sendT(getPlayer(), "{ce}You are no longer allowed to do that.");
+                    return;
+                }
+
+                if (newName == null || newName.isEmpty()) {
+                    I.sendT(getPlayer(), "{ce}Map names can't be empty.");
+                    return;
+                }
+                if (newName.equals(map.getName())) {
+                    return;
+                }
+
+                map.rename(newName);
+                I.sendT(getPlayer(), "{cs}Map successfully renamed.");
+
+                if (getParent() != null) {
+                    RunTask.later(() -> Gui.open(getPlayer(), this), 1L);
+
+                } else {
+                    close();
+                }
+            }, map.getName(), this);
+
+        } catch (IllegalStateException e) {
+            PluginLogger.info("error while renaming");
```

Comment:
`Error while renaming map` and include the stacktrace?

Reply:
Done I switched to Pluginlogger.error